docs/modules/fault-tolerance/pages/backups.adoc

```diff
@@ -7,6 +7,15 @@ The distribution happens on the partition level; the data and its backups are st
 memory partitions. See the xref:overview:data-partitioning.adoc[Data Partitioning] and xref:clusters:partition-group-configuration.adoc[Partition Grouping]
 for more information about the partitioning.
 
+[IMPORTANT]
+====
+Hazelcast does not support backup redistribution on a data structure level. Thus, changing the backup count of a nonempty data structure is **not** supported. For example, for a map with a nonzero backup count:
+
+1. increasing the backup countfootnote:change-backup-count[The backup count of a map can only be changed by modifying the xref:configuration:understanding-configuration.adoc#static-configuration[static configuration] and restarting the cluster.] does **not** create additional copies of existing map entries,
+2. decreasing the backup countfootnote:change-backup-count[] does **not** remove any backups of existing map entries, so some backups will have stale data if the map entries are overwritten, and
+3. if some backups are lost due to node failure, additional copies are **not** created on other nodes to meet the backup count.
```

Comment:
This is more complicated.

Eg. if you have 3 member cluster and want to increase number of backups from 1 to 2, you might just do a rolling restart with new config (disclaimer: I did not test it). The trick is that actual number of replicas (owner+backups) is <= cluster size. So if you have too few members, the cluster does not try create replicas which do not fit. But when a new member joins, replicas with higher indexes can be crated.

Also, if you were able to force at some point in all partitions to be owned by members with updated config, then subsequent migrations should use new backup count. One way of doing that is as above but shutting down some members. Another way might be by demoting and promoting some members, but you would have to make sure that each partition is "touched".

OFC, none of these scenarios is a supported way of doing that.

Reply:
when I was thinking more of it, it should actually be possible to change number of backups in static config by doing 2 rolling restarts: first for updating the config in members, second for (indirectly) triggering migrations. This is wasteful, because you need to send all data back and forth many times and the restart should be done ACTIVE cluster state, but should work.

this might not work with hotrestart if the static config is saved also in hot restart data (I did not check), and will be not possible with dynamic config (with or without hotrestart) because dynamic config is immutable

Reply:
Partition rebalancing might correct the number of backups as reported [here][1]. However, I am not sure whether we should accept this as a workaround and promote it in our documentation.

[1]: https://hazelcast.atlassian.net/browse/SUP-416?focusedCommentId=102524

Reply:
I am not sure if this procedure works by design or by accident. If we make sure that it really works, it might be provided by support if needed (and if is still works). But I would not add it to the docs as it does not seem to be tested automatically and could change in the future.